lib/fiware-iotagent-lib.js

```diff
@@ -116,6 +130,14 @@ function doActivate(newConfig, callback) {
         commandRegistry = require('./services/commands/commandRegistryMemory');
     }
 
+    if (checkNgsi2(newConfig)) {
+        logger.info(context, 'Using NGSIv2');
+        ngsi = ngsi2;
+        deviceService = deviceServiceNgsi2;
+        commands = commandsNgsi2;
+        subscriptions = subscriptionsNgsi2;
```

Comment:
Note IOTAs don't uses any subscription... this seems to be over-implemented. The three NGSIv2 interactions that should be supported are only the ones described in #527 

1. Registration to CB on device provision time.
2. Send updates to CB (for active attributes).
3. Acts as Context Provider for queries/updates coming from CB (for passive attributes and commands).

Of those, only point 2 is achiavable now (as 1 and 3 are not implemented/specified yet). I'd suggest to reduce the scope just to that point. That should reduce the size of the PR and make it reviewable.

Not sure of the NGSIv1-based code is structured, but if the POST /v1/updateContext is modularized in just one point (where the verb + url + payload is composed) maybe is a matter of just adapting that part, creating a NGSIv2 version for that and switching to the new/old method depending on the parameter in config.js.